### ISSUE TITLE
Search results updating multiple times without extra typing

### ISSUE THREAD

**Issue by plexish** [StampyAI/stampy-ui#67]
e.g. if you search:

how do i help

and wait long enough, it updated three times re-sorting the results one time, and twice on two other occasions. Updating once after typing ends is probably unavoidable, but more than once seems odd?

**Comment by ChrisRimmer** [comment]
Doing some investigation on this and while I can only get it to update the results list once after it opens, that still seems like one time too many.

So far I've got as far as working out that the first thing we see (just for a moment) is the list of results according to a plain text search, which quickly gets replaced by the tensorflow search result. I think we should defer displaying plain text results until tensorflow fails (which in fairness, I don't expect it would do often), but I don't know how tf error handling works. @Aprillion you got any ideas here?

**Comment by Aprillion** [comment]
The point of plain text search is to display "something" in the first few seconds while tensorflow was loading the ~30 MB model + in case something goes wrong and the external google model never loads.

I am proposing to delete the tensorflow - temporarily in #124 while we are discussing how to implement search in Google Docs architecture, but I am a proponent of BE search without client-side tensorflow ;) We still need to load a list of questions for other features down the line, so I wouldn't remove plain-text search yet, but integration of the 2 searches in more seamless way (or completely dropping the client-side plain-text search too) should wait until we finalize the new search for GDocs.

**Comment by plexish** [comment]
I just tested again and got two updates.

**Comment by ccstan99** [comment]
The debounce was supposed to help with this, but it was also causing the lag issues from #175. I dropped it down to 100ms from 500ms.

I also added the original userQuery along with the TF results. Maybe we can also keep a timestamp of when the results were last updated? Ideally, when receiving results from the tfWorker, we should compare the userQuery with current search input value and only display results if it hadn't been updated too recently, i.e. the timestamp was longer than some specified (100ms or 300ms?) ago.

My life & work outside of Stampy is ramping up, so I'm not as responsive as I wish to be and am afraid I might be blocking progress on this. If someone (@mruwnik?) with more availability can resolve the rest of this issue, that'd be super helpful.
